### ISSUE TITLE
Bash Traps, on-failure tasks

### ISSUE THREAD

**Issue by brad-jones** [rliebz/tusk#32]
I find my self doing this sort of thing:

```yaml
command: >-
  set -eo pipefail;
  function cleanup {
    rm -rf /foo/bar
  };
  trap cleanup EXIT;
  run commands that might fail...
```

Would be nice to be able to do something like:

```yaml
command:
  execute: foo
  on-failure: bar
```

**Comment by rliebz** [comment]
I've thought about this a bit, but it's good to see a use case other than what I've been doing.

One of the tricky things about any sort of "on failure" behavior is that there kind of is an on-failure behavior already—the program exits. So if `tusk` adds an `on-failure`, either it's a pre-exit behavior, which means there's no recovering, or it lets the user handle the error, in which case I'm not sure if there's a good way to avoid swallowing error context (the exit code). I'm also not sure that proper exception handling would even belong in a task runner.

If this pattern is typical of your use case, what do you think about instead having a `teardown`/`cleanup`/`after` behavior that runs whether or not commands were successful? Potentially:

```yaml
tasks:
  my-task:
    run: exit 1
    after: rm -rf /foo/bar
```

Where the properties of `after` could be the same as `run`, although the initial exit code is propagated to the command line.

While this syntax would limit cleanup to the task-level rather than the command-level, it would still be possible to have different cleanups for different steps:

```yaml
tasks:
  task-1:
    run: exit 1
    after: rm -rf /foo/1
  task-2:
    run: exit 2
    after: rm -rf /foo/2
  my-task:
    run:
      - task: task-1
      - task: task-2
    after: echo "All done" >> done.txt
```

And running it:
```
$ tusk my-task
Running: exit 1
Running: rm -rf /foo/1
Running: echo "All done" >> done.txt
exit status 1
```

**Comment by brad-jones** [comment]
Totally agree, its not so much `on-failure`, it's something that I want to do `after` regardless of the result of the previous command.

I guess it's similar to a `finally` clause, I don't want to `catch` the error, yes that (the exit code) should still propagate to the caller.

Per task would be fine.

**Comment by rliebz** [comment]
Resolved in [`v0.4.1`](https://github.com/rliebz/tusk/releases/tag/v0.4.1) with the [`finally` clause](https://rliebz.github.io/tusk/spec/#finally).

Thanks for the suggestion!
